### PR TITLE
[IMP] pivot: support multiple measures 

### PR DESCRIPTION
**Pull request by pro-odoo**
## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/side_panel/pivot/pivot_layout_configurator/pivot_dimension/pivot_dimension.ts [hunk 1]
```diff
@@ -27,6 +28,14 @@ export class PivotDimension extends Component<Props, SpreadsheetChildEnv> {
...
     onRemoved: { type: Function, optional: true },
+    onNameUpdated: { type: Function, optional: true },
     slots: { type: Object, optional: true },
   };
+
+  private state: { isUpdate: boolean } = useState({ isUpdate: false });
+
+  updateName(ev: InputEvent) {
+    this.props.onNameUpdated?.(this.props.dimension, (ev.target as HTMLInputElement).value);
+    this.state.isUpdate = false;
```
Comment (anhe-odoo) [inline src/components/side_panel/pivot/pivot_layout_configurator/pivot_dimension/pivot_dimension.ts:39]: What do you mean by `isUpdate` ? Maybe `isUpToDate` ?

Comment (pro-odoo) [reply src/components/side_panel/pivot/pivot_layout_configurator/pivot_dimension/pivot_dimension.ts:39]: I mean `I'm currently editing the title`. So no `isUpToDate`.
But i'm not convinced by `isUpdate` so if you have an idea...

Comment (anhe-odoo) [reply src/components/side_panel/pivot/pivot_layout_configurator/pivot_dimension/pivot_dimension.ts:39]: isUpdating ? Else I don't have any idea :/

File: src/components/side_panel/pivot/pivot_layout_configurator/pivot_dimension/pivot_dimension.xml [hunk 2]
```diff
@@ -4,7 +4,24 @@
       class="border py-1 px-2 d-flex flex-column shadow-sm pivot-dimension"
       t-att-class="{'bg-danger': !props.dimension.isValid}">
       <div class="d-flex flex-row justify-content-between align-items-center">
-        <span class="fw-bold" t-esc="props.dimension.displayName"/>
+        <div class="d-flex align-items-center">
+          <t t-if="state.isUpdate">
```
Comment (LucasLefevre) [inline src/components/side_panel/pivot/pivot_layout_configurator/pivot_dimension/pivot_dimension.xml:8]: what about being always in edit mode ? Just like in odoo. It has worked out pretty well.
It's less code and less friction for the user

**Review by LucasLefevre — COMMENTED** [review]
- I think pressing Enter after typing in the input should update the measure
- the input content is reset when clicking to move the cursor
[Screencast from 23-07-24 14:14:16.webm](https://github.com/user-attachments/assets/ddd7d11e-b175-47cf-b5c8-b1fdf33a55ab)

**Comment by pro-odoo** [comment]
> * I think pressing Enter after typing in the input should update the measure

Already done in the PR of custom name: https://github.com/odoo/o-spreadsheet/pull/4675
Sorry it wasn't clear that this PR depends on the task of the custom name :/

> * the input content is reset when clicking to move the cursor

I will test it in the other PR.

**Review by LucasLefevre — COMMENTED** [review]
"support multiple measures" looks good to me :+1:

**Review by LucasLefevre — APPROVED** [review]
robodoo r+

**Pull request closed without merging**